### PR TITLE
Initial bigtable/api example with automated builds. (#2)

### PR DESCRIPTION
**Pull request by coryan**
Create a simple program to list the Cloud Bigtable instances in a project, as well as the supporting `cmake` files, Travis build scripts, Dockerfiles to test with different distributions and compilers, and supporting documentation.

### CODE REVIEW

**Comment by SurferJeffAtGoogle** [comment]
The layout and organization of the code looks good.  Please add more build instructions to the README.  I failed to build:

```
rennie@rennie:~/gitrepos/cpp-docs-samples/bigtable/api$ cmake .
-- The CXX compiler identification is GNU 4.8.4
-- The C compiler identification is GNU 4.8.4
-- Check for working CXX compiler: /usr/bin/c++
-- Check for working CXX compiler: /usr/bin/c++ -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Check for working C compiler: /usr/bin/cc
-- Check for working C compiler: /usr/bin/cc -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Detecting C compile features
-- Detecting C compile features - done
-- Performing Test COMPILER_SUPPORTS_WERROR
-- Performing Test COMPILER_SUPPORTS_WERROR - Success
-- Performing Test COMPILER_SUPPORTS_WALL
-- Performing Test COMPILER_SUPPORTS_WALL - Success
-- Performing Test COMPILER_SUPPORTS_WX
-- Performing Test COMPILER_SUPPORTS_WX - Failed
-- Performing Test COMPILER_SUPPORTS_SWALL
-- Performing Test COMPILER_SUPPORTS_SWALL - Failed
-- Looking for pthread.h
-- Looking for pthread.h - found
-- Looking for pthread_create
-- Looking for pthread_create - not found
-- Looking for pthread_create in pthreads
-- Looking for pthread_create in pthreads - not found
-- Looking for pthread_create in pthread
-- Looking for pthread_create in pthread - found
-- Found Threads: TRUE  
-- Found Protobuf: /usr/local/lib/libprotobuf.so;-lpthread (found version "3.1.0") 
-- Found PkgConfig: /usr/bin/pkg-config (found version "0.26") 
-- Checking for module 'grpc++'
--   Found grpc++, version 1.1.0-dev
-- Checking for module 'grpc'
--   Found grpc, version 1.1.0-dev
-- Checking for module 'protobuf'
--   Found protobuf, version 3.1.0
You have called ADD_LIBRARY for library googleapis without any source files. This typically indicates a problem with your CMakeLists.txt file
-- Configuring done
CMake Error: Cannot determine link language for target "googleapis".
CMake Error: CMake can not determine linker language for target: googleapis
-- Generating done
-- Build files have been written to: /usr/local/google/home/rennie/gitrepos/cpp-docs-samples/bigtable/api
rennie@rennie:~/gitrepos/cpp-docs-samples/bigtable/api$ 
```

**Comment by coryan** [comment]
Thanks for the review.  I have updated the instructions as requested.

File: bigtable/api/README.md [hunk 1]
```diff
@@ -45,3 +45,27 @@ These samples demonstrate how to call the [Google Cloud Bigtable API](https://cl
...
+1.  **Generate googleapis gRPC source code.**
+    ```console
+    cd bigtable/api
+    env -u LANGUAGE make -C googleapis OUTPUT=$PWD/googleapis-gens
+    ```
+
+1.  **Compile the examples**
+    ```console
+    cmake .
+    make -j 2
```
Comment (SurferJeffAtGoogle) [inline bigtable/api/README.md]:
```
[ 99%] Building CXX object CMakeFiles/googleapis.dir/googleapis-gens/google/watcher/v1/watch.pb.cc.o
[ 99%] Linking CXX static library libgoogleapis.a
[ 99%] Built target googleapis
Scanning dependencies of target list_instances
[ 99%] Building CXX object CMakeFiles/list_instances.dir/list_instances.cc.o
/usr/local/google/home/rennie/gitrepos/cpp-docs-samples/bigtable/api/list_instances.cc: In function ‘int main(int, char**)’:
/usr/local/google/home/rennie/gitrepos/cpp-docs-samples/bigtable/api/list_instances.cc:55:27: error: ‘class grpc::Status’ has no member named ‘error_details’
                 << status.error_details() << std::endl;
                           ^
make[2]: *** [CMakeFiles/list_instances.dir/list_instances.cc.o] Error 1
make[1]: *** [CMakeFiles/list_instances.dir/all] Error 2
make: *** [all] Error 2
```

Comment (coryan) [inline bigtable/api/README.md]: Cool find, thanks.  I need your opinion on how to proceed here.  The `error_details` member function was introduced about 4 months ago:

https://github.com/grpc/grpc/commit/f4c8adddede470207fa46296d921592eda679cb1#diff-93bb8defdb4b5bfa42824ca7671638c3

It is part of the 1.4.2 release of gRPC:

https://github.com/grpc/grpc/blob/v1.4.2/include/grpc%2B%2B/impl/codegen/status.h

Would you prefer me to use an older version of gRPC?  If so, which one?  I have added a check to `CMakeList.txt` to verify the right version is installed.  If you think we should support all versions greater than X I will change the code to do that and change the checks too.

Comment (SurferJeffAtGoogle) [inline bigtable/api/README.md]: Is this happening due to an old version of something on my system?  Like protoc?
```
rennie@rennie:~/gitrepos/cpp-docs-samples/bigtable/api$ protoc --version
libprotoc 3.1.0
```

Comment (coryan) [inline bigtable/api/README.md]: Old version of grpc I believe.  You must have 1.4.1 or earlier.

Comment (SurferJeffAtGoogle) [inline bigtable/api/README.md]: How complicated would it be to detect the current version of gRPC and report a descriptive error message if it's too old?

**Pull request merged**